src/software/geom/voronoi_util.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+
+#include "software/geom/point.h"
+#include <boost/polygon/voronoi.hpp>
+#include "software/geom/rectangle.h"
+#include "software/geom/circle.h"
+
+using boost::polygon::voronoi_builder;
+using boost::polygon::voronoi_diagram;
+using boost::polygon::voronoi_vertex;
+
+/**
+ * Finds the points of intersection between the edges of a voronoi diagram and a bounding rectangle
+ *
+ * @param vd The voronoi diagram
+ * @param rec The bounding rectangle
+ * @param points The points used to create the Voronoi diagram.
+ * @return
+ */
+std::vector<Point> findVoronoiEdgeRecIntersects(const voronoi_diagram<double>& vd, Rectangle rec, std::vector<Point> points);
+
+/**
+ * From a Voronoi diagram, produce a set of empty circles.
```

Comment:
Hmm, maybe expand this a bit to "Find the set of open circles whose origin lies within the given rectangle, and that do not overlap any of the vertices in the Delaunay triangulation of the given voronoi diagram"